libopae/src/mmio.c

```diff
@@ -87,38 +88,39 @@ fpga_result __FPGA_API__ fpgaWriteMMIO32(fpga_handle handle,
 	return result;
 }
 
-fpga_result __FPGA_API__ fpgaReadMMIO32(fpga_handle handle,
-					uint32_t mmio_num,
-					uint64_t offset,
-					uint32_t *value)
+static fpga_result port_mmap_region(fpga_handle handle,
+			     void **vaddr,
+			     uint64_t size,
+			     uint32_t flags,
+			     uint64_t offset,
+			     uint32_t mmio_num)
 {
+	void *addr;
 	int err;
 	struct _fpga_handle *_handle = (struct _fpga_handle *) handle;
 	fpga_result result = FPGA_OK;
 
-	if (offset % sizeof(uint32_t) != 0) {
-		FPGA_MSG("Misaligned MMIO access");
-		return FPGA_INVALID_PARAM;
-	}
+	UNUSED_PARAM(mmio_num);
+
+	/* Assure returning pointer contains allocated memory */
+	ASSERT_NOT_NULL(vaddr);
 
 	result = handle_check_and_lock(_handle);
 	if (result)
 		return result;
 
-	struct wsid_map *wm = wsid_find_by_index(_handle->mmio_root, mmio_num);
-	if (!wm) {
-		FPGA_MSG("Trying to access MMIO before calling fpgaMapMMIO()");
-		result = FPGA_INVALID_PARAM;
-		goto out_unlock;
-	}
-
-	if (offset > wm->len) {
-		FPGA_MSG("offset out of bounds");
+	/* Map MMIO memory */
+	addr = (void *) mmap(NULL, size, flags, MAP_SHARED, _handle->fddev, offset);
+	if (vaddr == MAP_FAILED) {
```

Comment:
Is this correct?  addr is catching the result of mmap, yet vaddr is compared to MAP_FAILED.  I think addr should be compared..